src/osd/ECBackend.cc

```diff
@@ -1584,28 +1596,50 @@ int ECBackend::get_min_avail_to_read_shards(
 int ECBackend::get_remaining_shards(
   const hobject_t &hoid,
   const set<int> &avail,
+  const set<int> &want,
+  const read_result_t &result,
   set<pg_shard_t> *to_read,
   bool for_recovery)
 {
   assert(to_read);
 
   set<int> have;
   map<shard_id_t, pg_shard_t> shards;
+  set<pg_shard_t> error_shards;
+  for (auto &p : result.errors) {
+    error_shards.insert(p.first);
+  }
 
-  get_all_avail_shards(hoid, have, shards, for_recovery);
+  get_all_avail_shards(hoid, error_shards, have, shards, for_recovery);
+
+  map<int, vector<pair<int, int>>> need;
```

Comment:
@VictorDenisov This has to be adjusted for Luminous.  It should be 'set<int> need'.

Reply:
Luminous doesn't include ceph/ceph#15193